cmd/fleet/calendar_cron.go

```diff
@@ -365,6 +363,24 @@ func processFailingHostExistingCalendarEvent(
 	return nil
 }
 
+func shouldReloadCalendarEvent(now time.Time, calendarEvent *fleet.CalendarEvent, hostCalendarEvent *fleet.HostCalendarEvent) bool {
+	// Check the user calendar every 30 minutes (and not every cron run)
+	// to reduce load on both Fleet and the calendar service.
+	if time.Since(calendarEvent.UpdatedAt) > 30*time.Minute {
+		return true
+	}
+	// If the event is supposed to be happening now, we want to check if the user moved/deleted the
+	// event on the last minute.
+	if eventHappeningNow(now, calendarEvent) && hostCalendarEvent.WebhookStatus == fleet.CalendarWebhookStatusNone {
+		return true
+	}
+	return false
+}
+
+func eventHappeningNow(now time.Time, calendarEvent *fleet.CalendarEvent) bool {
+	return now.After(calendarEvent.StartTime) && now.Before(calendarEvent.EndTime)
```

Comment:
```suggestion
	return !now.Before(calendarEvent.StartTime) && now.Before(calendarEvent.EndTime)
```

To catch when now == StartTime